### ISSUE TITLE
Syntax analysis is not re-executed in the IDE diagnostic service on adding global suppressions

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#6007]
1. Create a new C# class library with below code:
    ```C#
    class Class1
    {
        // Comment with trailing whitespaces     
    }
    ```

2. Install [codecracker](http://www.nuget.org/packages/codecracker/1.0.0-rc3) analyzer nuget package (check "Include prerelease" checkbox in Nuget package manager).

3. Change rule _"CC0065 (Remove trailing whitespace)"_ to be a warning from the ruleset editor.

4. Build project and see CC0065 diagnostic:
    ```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CC0065	Remove trailing whitespace	ClassLibrary14	c:\users\mavasani\documents\visual studio 2015\Projects\ClassLibrary14\ClassLibrary14\Class1.cs	3	

    ```

5. From the lightbulb menu select "Suppress CC0065" -> "In Suppression File" and hit enter

6. Verify suppression added to GlobalSuppressions.cs

  **Expected:** Diagnostic is suppressed in the error list + squiggle disappears.
  **Actual:** Diagnostic is retained in the error list and squiggle also doesn't go away. If you rebuild the project, the output window doesn't show the original diagnostic (though it shows 2 more instances of CC0065 in the newly added GlobalSuppressions file), but the error list still shows the origin CC0065 intellisense diagnostic.

7. Edit Class1.cs at an unrelated location - say hit enter at the end of the last line of the source file.
You will now see the expected behavior, and the diagnostic and squiggle disappear.

**Details:** The reason for this bug is that CC0065 is a syntax diagnostic (syntax tree analyzer action), and our IDE diagnostic service assumes that syntax diagnostics only depend on text version of the document. Adding a global suppression to other file invalidates this assumption. We cannot run all syntax analyzers on every single edit, that would cause perf problems - we need to figure out how to implement this reasonably.

**Comment by mavasani** [comment]
/cc @heejaechang @morrisjoe @srivatsn 

This will cause bulk suppress syntax diagnostics in error list to persist as active, even though we added global suppressions for them.

**Comment by heejaechang** [comment]
basically, allowing this suppression will make all of syntax based analyzer to be same cost as semantic based analyzer. and I don't think there is any way to get around it if we support this.

**Comment by srivatsn** [comment]
Can we have the suppression codefix request a reanalysis and so only for those cases, it's costly but the normal typing scenarios aren't affected.

**Comment by heejaechang** [comment]
@srivatsn what about people manually typing suppression. due to cache and manual typing, I think what Manish suggested is only simple solution.

otherwise, we can go a bit more complex one which involves caching last suppress attributes in the solution per project and whenever there is a top level change, we diff them to see whether we need to re-analyze projects and create a new version (suppressed attribute version or something) and use that also in syntax version or project version.

**Comment by mavasani** [comment]
@heejaechang We decided in triage meeting that lets do the minimal change for Update1 for the suppression via code fix scenario. Post update1 we can tackle this more generically to handle manual edits - that needs lot more perf validations to happen.
